### PR TITLE
ApiPlugin typings added

### PR DESCRIPTION
**Pull request by Legends**
**What kind of change does this PR introduce?**
ApiPlugin typings added

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
no

### CODE REVIEW

File: lib/APIPlugin.js [hunk 1]
```diff
@@ -30,8 +30,18 @@ const REPLACEMENT_TYPES = {
 	__webpack_nonce__: "string"
 };
 /* eslint-enable camelcase */
+/** @typedef {import("./Parser")} Parser */
```
Comment (ooflorent) [inline lib/APIPlugin.js]: Could you move this right after the `require` calls?

Comment (Legends) [reply lib/APIPlugin.js]: Of course np. I followed the pattern of Sean in `chunk.js`, he moved the typings after "all" `const` decalarations..
Will update it

Comment (Legends) [reply lib/APIPlugin.js]: done

File: lib/APIPlugin.js [hunk 2]
```diff
@@ -31,7 +33,16 @@ const REPLACEMENT_TYPES = {
...
 /* eslint-enable camelcase */
 
+/**
+ * Make __webpack_public_path__, __webpack_require__, __webpack_modules__, __webpack_chunk_load__ accessible.
+ * Ensures that require.valueOf and require.onError are not processed by other plugins.
+ */
 class APIPlugin {
+	/**
+	 * Apply the plugin
+	 * @param {TODO} compiler Webpack Compiler
```
Comment (sokra) [inline lib/APIPlugin.js]: TODO -> Compiler

File: lib/APIPlugin.js [hunk 3]
```diff
@@ -40,7 +41,7 @@ const REPLACEMENT_TYPES = {
 class APIPlugin {
 	/**
 	 * Apply the plugin
-	 * @param {TODO} compiler Webpack Compiler
+	 * @param {Compiler} compiler Webpack Compiler
 	 * @returns {void}
```
Comment (Legends) [inline lib/APIPlugin.js]: If I remove the `@returns {void}` as defined in the typings guide, I get:

image

File: lib/APIPlugin.js [hunk 4]
```diff
@@ -42,7 +42,6 @@ class APIPlugin {
 	/**
 	 * Apply the plugin
 	 * @param {Compiler} compiler Webpack Compiler
-	 * @returns {void}
```
Comment (Legends) [inline lib/APIPlugin.js]: https://github.com/webpack/webpack/issues/7122 - section Return!?

File: lib/APIPlugin.js [hunk 5]
```diff
@@ -56,7 +55,6 @@ class APIPlugin {
 				/**
 				 *
 				 * @param {Parser} parser webpack parser
-				 * @returns {void}
```
Comment (Legends) [inline lib/APIPlugin.js]: https://github.com/webpack/webpack/issues/7122 - section Return!?

Comment (sokra) [reply lib/APIPlugin.js]: hmm yes the guide was wrong `@returns {void}` can't be omitted. Omitting it would mean `@returns {any}`.

Comment (Legends) [reply lib/APIPlugin.js]: ok, will have to add them back then... in all PR's. This one is done.

**Pull request closed without merging**